src/Middleware/Diagnostics/src/Microsoft.AspNetCore.Diagnostics.csproj

```diff
@@ -18,6 +18,7 @@
     <Reference Include="Microsoft.AspNetCore.Diagnostics.Abstractions" />
     <Reference Include="Microsoft.AspNetCore.Hosting.Abstractions" />
     <Reference Include="Microsoft.AspNetCore.Http.Extensions" />
+    <Reference Include="Microsoft.AspNetCore.Routing" />
```

Comment:
Is this routing or routing.abstractions?

Reply:
Routing. It is needed for RouteEndpoint and RoutePattern.